services/search/pkg/config/defaults/defaultconfig.go

```diff
@@ -68,6 +68,14 @@ func EnsureDefaults(cfg *config.Config) {
 	if cfg.MachineAuthAPIKey == "" && cfg.Commons != nil && cfg.Commons.MachineAuthAPIKey != "" {
 		cfg.MachineAuthAPIKey = cfg.Commons.MachineAuthAPIKey
 	}
+
+	if cfg.Reva == nil && cfg.Commons != nil && cfg.Commons.Reva != nil {
```

Comment:
Can we change this to 
```suggestion
	if cfg.Reva.Address == "" && cfg.Commons != nil && cfg.Commons.Reva != nil {
```

Then we wouldn't need to change it to a `*Reva` and also we get rid of the `else if` statement.

Reply:
I copied this from other occurences in the codebase, assuming the pointer was necessary for correct config merging ... maybe @wkloucek or @dragonchaser can shed some light on this.

Reply:
I still would check for nil beforehand...
```
if cfg.Reva == nil && cfg.Reva.Address == "" && cfg.Commons != nil && cfg.Commons.Reva != nil {
```

Reply:
for consistency we should use the same approach everywhere like: https://github.com/owncloud/ocis/blob/master/services/proxy/pkg/config/defaults/defaultconfig.go#L183 so @butonic change makes sense to me since the search service is the only service so far that does not use this variable as a pointer (I think it was merged in paralell to our refactoring).

Reply:
Ok. Sounds good to me. Thanks for clarifiying!